### PR TITLE
Fix dict files once and for all

### PR DESCRIPTION
**Pull request by klshuster**
Fixes DrQA nightly gpu test. Also, fixes a long-standing issue of moving a dictionary.

Tested via running DrQA test locally.

### CODE REVIEW

File: parlai/core/agents.py [hunk 1]
```diff
@@ -391,9 +391,17 @@ def load_agent_module(opt):
             if k not in new_opt:
                 new_opt[k] = v
         new_opt['model_file'] = model_file
-        if new_opt.get('dict_file') and not os.path.isfile(new_opt['dict_file']):
-            raise RuntimeError('WARNING: Dict file does not exist, check to make '
-                               'sure it is correct: {}'.format(new_opt['dict_file']))
+        if not new_opt.get('dict_file'):
+            new_opt['dict_file'] = model_file + '.dict'
+        elif new_opt.get('dict_file') and not os.path.isfile(new_opt['dict_file']):
```
Comment (jaseweston) [inline parlai/core/agents.py]: i was doing this the other way round.. if you have a   model.dict then it just uses it, overriding whatever is in the dict_file .. is your way better?

Comment (klshuster) [reply parlai/core/agents.py]: Your way probably makes more sense in practice, but this way at least respects the case where someone explicitly specifies a `--dict_file` on the command line (which, in your way, we would ignore)

**Comment by jaseweston** [comment]
yea should be fine

**Comment by stephenroller** [comment]
Land this but let's talk about forceful dicts at the next meeting.

**Pull request merged**